doc/source/ray-air/examples/batch_tuning.ipynb

```
@@ -0,0 +1,1197 @@
+    "\n",
+    "**Below, we define our search space is:**\n",
+    "- 2 different Scikit-learn algorithms \n",
+    "- Some or all NYC taxi drop-off locations. \n",
+    "\n",
+    "**And we define the search algorithm is:**\n",
+    "- Grid search.\n",
+    "\n",
+    "> This means every permutation of each algorithm and each NYC Taxi drop-off location will be a separate experiment!  \n",
+    "\n",
+    "Ray Tune partitions the Search space using the specified Search algorithm and takes care of running a Tune job on each partition in parallel.  Specifically, Ray Tune will pass a config dictionary to each partition and make a Trainable function call."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "id": "7de03814-4eec-4363-a623-790f040b99be",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# 1. Define a search space.\n",
+    "sample_locations = [145, 166, 152] if SMOKE_TEST else all_location_ids\n",
+    "search_space = {\n",
+    "    \"model\": tune.grid_search([LinearRegression(fit_intercept=True), \n",
+    "                                DecisionTreeRegressor(max_depth=3)]),\n",
+    "    \"location\": tune.grid_search(sample_locations),\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "28814e79-f53e-4527-a959-e952f7d57fe5",
+   "metadata": {},
+   "source": [
+    "## Define a Trainable (callable) function <a class=\"anchor\" id=\"define_trainable\"></a>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "cdb3c9a4-b2df-4063-8c04-f5e0136c90bf",
+   "metadata": {},
+   "source": [
+    "🧪 Typically when you are running Data Science experiments, you want to be able to keep track of summary metrics for each experiment, so you can decide at the end which experiments were the best.  That way, you can decide which model to deploy.\n",
```

let's make sure to start off with what's relevant to the workload here (we define the trainable in order to train and evaluate a sklearn model on a data partition) and then move to details about metrics (because we want to report the error and checkpoint the model for reuse)